extract_metadata.py

```diff
@@ -0,0 +1,111 @@
+                        break
+
+                if subject == 'Unknown' and any(filename in asset_split[-1].lower() for filename in ['dataset_description.json', 'participants.tsv', 'readme.md', 'samples.tsv']):
+                    subject = 'n/a'
+
+                modality = next((value for key, value in modalities.items() 
+                                if key in asset_split[-1].lower()), 
+                                'Unknown')
+
+                suffix = Path(asset.path).suffixes[0][1:] if Path(asset.path).suffixes else ''
+
+                df.loc[len(df)] = [latest_dataset.identifier,
+                                    latest_dataset.version.identifier,
+                                    subject,
+                                    modality,
+                                    asset.path, 
+                                    asset_split[-1],
+                                    suffix,
+                                    asset_split[0],
+                                    metadata_dict['contentSize']]
+
+    return df
+
+# Summarize data across datasets
+def summarize_datasets(df):
+    df_datasets = pd.DataFrame(columns=["Dataset",
+                                        "Modality",
+                                        "Size (GB)",
+                                        "Subject", 
+                                        "Extension"])
+
+    for dataset in df['Dataset'].unique():
+        df_datasets.loc[len(df_datasets)] = [dataset,
+                    ','.join(df[(df['Dataset'] == dataset)]['Modality'].unique()),
+                    round(sum(df[(df['Dataset'] == dataset)]['Size (bytes)'])/(1000**3),2),
+                    ','.join(df[(df['Dataset'] == dataset)]['Subject'].unique()),
+                    ','.join(df[(df['Dataset'] == dataset)]['Extension'].unique())]
+
+    return df_datasets
+
+# Summarize data across modalities
+def summarize_modalities(df):
+    local_modalities = modalities.copy()
+    local_modalities['unknown'] = 'Unknown'
+
+    df_modalities = pd.DataFrame(columns=["Modality",
+                                    "Size (GB)",
+                                    "Subject", 
+                                    "Extension"])
+    for _, value in local_modalities.items():
+        df_modalities.loc[len(df_modalities)] = [value,
+                    round(sum(df[(df['Modality'] == value)]['Size (bytes)'])/(1000**3),2),
+                    ','.join(df[(df['Modality'] == value)]['Subject'].unique()),
+                    ','.join(df[(df['Modality'] == value)]['Extension'].unique())]
+
+    return df_modalities
+
+if __name__ == "__main__":
+    df = extract_assets()
+    df_datasets = summarize_datasets(df)
+    df_modalities = summarize_modalities(df)
+
+    df.to_csv('./lincbrain_assets.csv', index=False)
+    df_datasets.to_csv('./lincbrain_datasets.csv', index=False)
+    df_modalities.to_csv('./lincbrain_modalities.csv', index=False)
```

Comment:
Just as an FYI -- if you wanted to be more efficient here, you could make the `.csv` a temp. file, export the temp. file, and then delete it -- if we were working in production at scale, we most likely would take that approach.

https://docs.python.org/3/library/tempfile.html

Reply:
That's good to know.  Thanks.